pandas/core/dtypes/missing.py

```diff
@@ -506,14 +506,31 @@ def array_equals(left: ArrayLike, right: ArrayLike) -> bool:
         return array_equivalent(left, right, dtype_equal=True)
 
 
-def infer_fill_value(val):
+def infer_fill_value(val, length: int):
     """
     infer the fill value for the nan/NaT from the provided
     scalar/ndarray/list-like if we are a NaT, return the correct dtyped
     element to provide proper block construction
     """
     if not is_list_like(val):
         val = [val]
+
+    if type(val).__name__ == "PandasArray":
```

Comment:
this is awful, but the relevant tests patch PandasArray._typ so that `isinstance(val, ABCPandasArray)` returns False

Reply:
umm why can't we patch *this* routine for that test then rather than doing this?

Reply:
good idea, updated.  we might be able to get rid of some other kludges the same way